pages/studentPortal/web/index.tsx

```diff
@@ -42,7 +44,7 @@ export default function StudentPortalPage() {
       />
       <div className="grid grid-cols-1 gap-4">
         {units.map((it, index) => (
-          <UnitView key={index} data={it} />
+          <UnitView key={index} data={it} userRole={userRole} />
```

Comment:
You did a great job implementing this feature but we should discuss higher level code design.

An important principle for you to understand is the single responsibility principle.

You want each component to be responsible for just one thing. You want to try and separate your UI code from the logic code.

The unit view is a UI component that should only have to worry about whether to render itself as locked, enabled, grayed out etc...

The UI should not need to know anything about the currently logged in user.

Your implementation requires the unitview depend on the current users role.

Let's discuss how we can change your code design so your UI component doesn't depend on user role.

Reply:
This makes a lot of sense.  
I have removed the userRole from all the components, and check it in the index.tsx as you suggest and then deploy a freemium or a regular experience depending on the role as called on that page.

I moved the curriculum/business logic to a freemiumUnits.ts file so that this controls the units that we want to include or not include and deploys them the same way to maintain consistency between how we are doing things with the freemium users and the full users.

![image](https://user-images.githubusercontent.com/111075855/224585605-62e541e3-f9f9-452d-89c9-87396c95f4e6.png)

This is one solution, but I would love to chat to see other ideas that you would suggest to either scrap this approach or improve on it!